File: remove-duplicates-from-sorted-array/removeDuplicates.go
func removeDuplicates(nums []int) int {
    previous, result, pointer := math.MaxInt32, 0, 0
    
    for _, v := range nums {
        if(v != previous) {
            nums[pointer] = v
            previous = v
            pointer++
            result++
        }
    }

    return result
}

File: remove-duplicates-from-sorted-array/RemoveDuplicates.cs
public class Solution
{
    public int RemoveDuplicates(int[] nums)
    {
        var result = 0;
        var pointer = 0;
        var previous = int.MinValue;

        foreach (var num in nums)
        {
            if (num != previous)
            {
                nums[pointer++] = num;
                previous = num;
                result++;
            }
        }

        return result;
    }
}